### PR TITLE
NOSTORY:removed redundant code.

### PR DESCRIPTION
**Pull request by j0hnmwangi**
# Description

Quick fix for redundant code that forces a reconfiguration of customer-managed policies when doing a second terraform apply.

## Features/Fixes/Patches/Changes list:

Please check the [CHANGELOG.md](https://github.com/boldlink/<REPOSITORY>/blob/fixes/iam-role/CHANGELOG.md#103---2022-06-29)

### PR Owners Checklist:
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you checked there aren't open [Pull Requests](../../../pulls) of upstream or parallel code that will cause conflicts?
* [x] Have you updated the `CHANGELOG.md`?
* [x] Have you updated the `README.md`(s)?
* [x] OWNER: Does your submission pass?
    * [x] Pre-commit (attach logs/print screen to this PR)
    * [x] Checkov/terrascan (attach logs/print screen to this PR)
    * [x] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### PR Reviewers Checklists?
* [ ] Are all your comments/changes resolved?
* [ ] Are you happy with the OWNER checklists and additional information provided?
* [ ] Does your review tests pass?
    * [ ] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?

### Checkov logs
[checkov_logs.txt](https://github.com/boldlink/terraform-aws-iam-role/files/9008202/checkov_logs.txt)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,13 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+
+## [1.0.3] - 2022-06-29
+### Description
+- Fix: Removed redundant code causing changes when doing a second
+       terraform apply in customer-managed policies.
```
Comment (hugoalmeida264) [inline CHANGELOG.md]: Make it a single line comment

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,8 +27,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Feature: IAM role Policy attachment
 - Feature: IAM role Policy
 - Feature: IAM Role
-- ...
 
+[Unreleased]: https://github.com/boldlink/terraform-aws-iam-role/releases/tag/1.0.3...HEAD
```
Comment (hugoalmeida264) [inline CHANGELOG.md:31]: links in markdown are `[text](link)`

File: main.tf [hunk 3]
```diff
@@ -25,17 +25,10 @@ resource "aws_iam_policy" "main" {
...
 
 resource "aws_iam_role_policy_attachment" "main" {
   count      = var.policy != null ? 1 : 0
   role       = aws_iam_role.main.name
   policy_arn = join("", aws_iam_policy.main.*.arn)
 }
-
-resource "aws_iam_role_policy_attachment" "managed" {
-  count      = length(var.managed_policy_arns) > 0 ? length(var.managed_policy_arns) : 0
-  role       = aws_iam_role.main.name
```
Comment (hugoalmeida264) [inline main.tf:39]: Wouldn't this be a good thing to maintain? you might want to attach these policies in the future.

**Review by hugoalmeida264 — CHANGES_REQUESTED** [review]

**Review by patrickmukumbu — CHANGES_REQUESTED** [review]
Since the complete example is showing usage of inline_policy, I think it would be nice to remove the third example (`with_inline_policy`)
@ndegwajohn

File: main.tf (hunk 3 above)
Comment (j0hnmwangi) [reply main.tf:39]: What the resource aimed to achieve was already being done by the `managed_policy_arns` argument. The aws-managed policies can be added directly to the list..

**Review by patrickmukumbu — APPROVED** [review]

**Review by hugoalmeida264 — APPROVED** [review]

**Pull request merged**